wgpu-hal/src/gles/fence.rs

```diff
@@ -0,0 +1,167 @@
+use std::sync::atomic::Ordering;
+
+use glow::HasContext;
+
+use crate::AtomicFenceValue;
+
+#[derive(Debug, Copy, Clone)]
+struct GLFence {
+    sync: glow::Fence,
+    value: crate::FenceValue,
+}
+
+#[derive(Debug)]
+pub struct Fence {
+    last_completed: AtomicFenceValue,
+    pending: Vec<GLFence>,
+    fence_mode: wgt::GlFenceBehavior,
+}
+
+impl crate::DynFence for Fence {}
+
+#[cfg(send_sync)]
+unsafe impl Send for Fence {}
+#[cfg(send_sync)]
+unsafe impl Sync for Fence {}
+
+impl Fence {
+    pub fn new(options: &wgt::GlBackendOptions) -> Self {
+        Self {
+            last_completed: AtomicFenceValue::new(0),
+            pending: Vec::new(),
+            fence_mode: options.short_circuit_fences,
+        }
+    }
+
+    pub fn signal(
+        &mut self,
+        gl: &glow::Context,
+        value: crate::FenceValue,
+    ) -> Result<(), crate::DeviceError> {
+        if self.fence_mode.is_auto_finish() {
+            *self.last_completed.get_mut() = value;
+            return Ok(());
+        }
+
+        let sync = unsafe { gl.fence_sync(glow::SYNC_GPU_COMMANDS_COMPLETE, 0) }
+            .map_err(|_| crate::DeviceError::OutOfMemory)?;
+        self.pending.push(GLFence { sync, value });
+
+        Ok(())
+    }
+
+    pub fn satisfied(&self, value: crate::FenceValue) -> bool {
+        self.last_completed.load(Ordering::Relaxed) >= value
```

Comment:
I can see this is just moved code, but intuitively I'd expect acquire semantics here (and release at the end of `get_latest`/`wait`) since I'd expect to see all changes prior to the last change of `last_completed`

Reply:
Yeah that's weird, I'm going to change them to acquire/release just to be sure.